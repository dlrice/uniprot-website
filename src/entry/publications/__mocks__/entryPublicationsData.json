{
  "facets": [
    {
      "label": "Scale",
      "name": "scale",
      "allowMultipleSelection": false,
      "values": [
        {
          "value": "Small",
          "count": 2272
        },
        {
          "value": "Large",
          "count": 116
        }
      ]
    }
  ],
  "results": [
    {
      "id": "1",
      "literatureEntry": {
        "pubmedId": 2881207,
        "doiId": "10.1038/325733a0",
        "title": "The precursor of Alzheimer's disease amyloid A4 protein resembles a cell- surface receptor.",
        "authors": [
          "Kang J.",
          "Lemaire H.-G.",
          "Unterbeck A.",
          "Salbaum J.M.",
          "Masters C.L.",
          "Grzeschik K.-H.",
          "Multhaup G.",
          "Beyreuther K.",
          "Mueller-Hill B."
        ],
        "completeAuthorList": true,
        "publicationDate": "1987",
        "journal": "Nature",
        "firstPage": "733",
        "lastPage": "736",
        "volume": "325",
        "literatureAbstract": "Alzheimer's disease is characterized by a widespread functional disturbance of the human brain. Fibrillar amyloid proteins are deposited inside neurons as neurofibrillary tangles and extracellularly as amyloid plaque cores and in blood vessels. The major protein subunit (A4) of the amyloid fibril of tangles, plaques and blood vessel deposits is an insoluble, highly aggregating small polypeptide of relative molecular mass 4,500. The same polypeptide is also deposited in the brains of aged individuals with trisomy 21 (Down's syndrome). We have argued previously that the A4 protein is of neuronal origin and is the cleavage product of a larger precursor protein. To identify this precursor, we have now isolated and sequenced an apparently full-length complementary DNA clone coding for the A4 polypeptide. The predicted precursor consists of 695 residues and contains features characteristic of glycosylated cell-surface receptors. This sequence, together with the localization of its gene on chromosome 21, suggests that the cerebral amyloid deposited in Alzheimer's disease and aged Down's syndrome is caused by aberrant catabolism of a cell- surface receptor.",
        "statistics": {
          "reviewedProteinCount": 1,
          "unreviewedProteinCount": 0,
          "mappedProteinCount": 0
        }
      },
      "uniProtReference": {
        "citation": {
          "citationType": "journal article",
          "authors": [
            "Kang J.",
            "Lemaire H.-G.",
            "Unterbeck A.",
            "Salbaum J.M.",
            "Masters C.L.",
            "Grzeschik K.-H.",
            "Multhaup G.",
            "Beyreuther K.",
            "Mueller-Hill B."
          ],
          "citationXrefs": [
            {
              "databaseType": "PubMed",
              "id": "2881207"
            },
            {
              "databaseType": "DOI",
              "id": "10.1038/325733a0"
            }
          ],
          "title": "The precursor of Alzheimer's disease amyloid A4 protein resembles a cell-surface receptor.",
          "publicationDate": "1987",
          "journal": "Nature",
          "firstPage": "733",
          "lastPage": "736",
          "volume": "325"
        },
        "referencePositions": ["NUCLEOTIDE SEQUENCE [MRNA] (ISOFORM APP695)"],
        "referenceComments": [
          {
            "value": "Brain",
            "type": "TISSUE"
          }
        ]
      },
      "categories": ["Sequence", "Sequence"],
      "publicationSource": "Swiss-Prot"
    },
    {
      "id": "2",
      "literatureEntry": {
        "pubmedId": 10097139,
        "doiId": "10.1073/pnas.96.7.3922",
        "title": "Constitutive and regulated alpha-secretase cleavage of Alzheimer's amyloid precursor protein by a disintegrin metalloprotease.",
        "authors": [
          "Lammich S.",
          "Kojro E.",
          "Postina R.",
          "Gilbert S.",
          "Pfeiffer R.",
          "Jasionowski M.",
          "Haass C.",
          "Fahrenholz F."
        ],
        "completeAuthorList": true,
        "publicationDate": "1999",
        "journal": "Proc. Natl. Acad. Sci. U.S.A.",
        "firstPage": "3922",
        "lastPage": "3927",
        "volume": "96",
        "literatureAbstract": "Amyloid beta peptide (Abeta), the principal proteinaceous component of amyloid plaques in brains of Alzheimer's disease patients, is derived by proteolytic cleavage of the amyloid precursor protein (APP). Proteolytic cleavage of APP by a putative alpha-secretase within the Abeta sequence precludes the formation of the amyloidogenic peptides and leads to the release of soluble APPsalpha into the medium. By overexpression of a disintegrin and metalloprotease (ADAM), classified as ADAM 10, in HEK 293 cells, basal and protein kinase C-stimulated alpha-secretase activity was increased severalfold. The proteolytically activated form of ADAM 10 was localized by cell surface biotinylation in the plasma membrane, but the majority of the proenzyme was found in the Golgi. These results support the view that APP is cleaved both at the cell surface and along the secretory pathway. Endogenous alpha-secretase activity was inhibited by a dominant negative form of ADAM 10 with a point mutation in the zinc binding site. Studies with purified ADAM 10 and Abeta fragments confirm the correct alpha-secretase cleavage site and demonstrate a dependence on the substrate's conformation. Our results provide evidence that ADAM 10 has alpha-secretase activity and many properties expected for the proteolytic processing of APP. Increases of its expression and activity might be beneficial for the treatment of Alzheimer's disease.",
        "statistics": {
          "reviewedProteinCount": 1,
          "unreviewedProteinCount": 0,
          "mappedProteinCount": 8
        }
      },
      "literatureMappedReference": {
        "uniprotAccession": "P05067",
        "source": "Reactome",
        "sourceId": "R-HSA-9010028",
        "annotation": "Pathway"
      },
      "categories": ["Function"],
      "publicationSource": "Computationally mapped"
    },
    {
      "id": "3",
      "uniProtReference": {
        "citation": {
          "citationType": "journal article",
          "authors": ["Borg J.-P.", "Ooi J.", "Levy E.", "Margolis B."],
          "citationXrefs": [
            {
              "databaseType": "PubMed",
              "id": "8887653"
            },
            {
              "databaseType": "DOI",
              "id": "10.1128/mcb.16.11.6229"
            }
          ],
          "title": "The phosphotyrosine interaction domains of X11 and FE65 bind to distinct sites on the YENPTY motif of amyloid precursor protein.",
          "publicationDate": "1996",
          "journal": "Mol. Cell. Biol.",
          "firstPage": "6229",
          "lastPage": "6241",
          "volume": "16"
        },
        "referencePositions": [
          "INTERACTION WITH APBA1 AND APBB1",
          "MUTAGENESIS OF TYR-728; TYR-757; ASN-759 AND TYR-762"
        ]
      },
      "categories": ["Interaction", "Pathol"],
      "publicationSource": "Swiss-Prot"
    }
  ]
}
