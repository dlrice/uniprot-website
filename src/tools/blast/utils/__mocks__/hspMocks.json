{
  "longQueryMiddleHit": {
    "hsp": {
      "hsp_num": 2,
      "hsp_score": 4058,
      "hsp_bit_score": 1567.75,
      "hsp_expect": 0,
      "hsp_align_len": 770,
      "hsp_identity": 100,
      "hsp_positive": 100,
      "hsp_gaps": 0,
      "hsp_query_frame": "0",
      "hsp_hit_frame": "0",
      "hsp_strand": "none/none",
      "hsp_query_from": 771,
      "hsp_query_to": 1540,
      "hsp_hit_from": 1,
      "hsp_hit_to": 770,
      "hsp_qseq": "MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN",
      "hsp_mseq": "MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN",
      "hsp_hseq": "MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN"
    },
    "result": {
      "querySegments": [
        {
          "start": 1,
          "end": 771,
          "shape": "line",
          "color": "#014371",
          "opacity": 1
        },
        { "start": 771, "end": 1540, "color": "#014371", "opacity": 1 },
        {
          "start": 1540,
          "end": 2310,
          "shape": "line",
          "color": "#014371",
          "opacity": 1
        }
      ],
      "hitSegments": [
        {
          "start": 770,
          "end": 771,
          "shape": "line",
          "color": "#014371",
          "opacity": 1
        },
        { "start": 771, "end": 1540, "color": "#014371", "opacity": 1 },
        {
          "start": 1540,
          "end": 1540,
          "shape": "line",
          "color": "#014371",
          "opacity": 1
        }
      ]
    }
  },
  "shortQueryLongHit": {
    "hsp": {
      "hsp_num": 2,
      "hsp_score": 4058,
      "hsp_bit_score": 1567.75,
      "hsp_expect": 0,
      "hsp_align_len": 60,
      "hsp_identity": 100,
      "hsp_positive": 100,
      "hsp_gaps": 0,
      "hsp_query_frame": "0",
      "hsp_hit_frame": "0",
      "hsp_strand": "none/none",
      "hsp_query_from": 20,
      "hsp_query_to": 24,
      "hsp_hit_from": 40,
      "hsp_hit_to": 44,
      "hsp_qseq": "ABCDE",
      "hsp_mseq": "ABCDE",
      "hsp_hseq": "ABCDE"
    },
    "result": {
      "querySegments": [
        {
          "color": "#014371",
          "start": 20,
          "end": 40,
          "opacity": 1,
          "shape": "line"
        },
        { "color": "#014371", "end": 44, "opacity": 1, "start": 40 },
        {
          "color": "#014371",
          "start": 44,
          "end": 50,
          "opacity": 1,
          "shape": "line"
        }
      ],
      "hitSegments": [
        {
          "color": "#014371",
          "start": 1,
          "end": 40,
          "opacity": 1,
          "shape": "line"
        },
        { "color": "#014371", "end": 44, "opacity": 1, "start": 40 },
        {
          "color": "#014371",
          "start": 44,
          "end": 60,
          "opacity": 1,
          "shape": "line"
        }
      ]
    }
  }
}
