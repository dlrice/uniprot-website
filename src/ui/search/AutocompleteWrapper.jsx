import React, { Component, Fragment } from 'react';
import { Autocomplete } from 'franklin-sites';
// import withData from '../hoc/withData';

class AutocompleteWrapper extends Component {
  constructor(props) {
    super(props);
    // this.handleSelect = this.handleSelect.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange = (value) => {
    console.log(value);
  }

  render() {
    return (
      <Autocomplete
        data={[]}
        // onSelect={e => this.handleSelect(e)}
        onChange={v => this.handleChange(v)}
      />
    );
  }
}

export default AutocompleteWrapper;
// export default withData(url, appendUniqueValue)(EvidenceField);