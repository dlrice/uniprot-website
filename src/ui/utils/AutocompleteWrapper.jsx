// @flow
import React, { Component } from 'react';
import { Autocomplete } from 'franklin-sites';
import fetchData from './fetchData';
import appendUniqueId from './prepareData';
import { getSuggesterUrl } from '../apiUrls';

type Props = {
  url: string,
  onSelect: Function,
};

type Suggestions = {
  dictionary: string,
  query: string,
  suggestions: Array<string>,
};

type State = {
  data: Array<Suggestions>,
};

type Chosen = {
  id: string,
  itemLabel: string,
  pathLabel: string,
};

class AutocompleteWrapper extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { data: [] };
  }

  handleChange = (textInputValue: string) => {
    const { url } = this.props;
    const suggesterUrl = getSuggesterUrl(url, textInputValue);
    this.fetchOptions(suggesterUrl);
  };

  fetchOptions = (url: string) => {
    fetchData(url)
      // Remove duplicates for now as there is a bug in the API
      .then(data => [...new Set(data.data.suggestions)])
      .then(data => data.map(x => ({ pathLabel: x, itemLabel: x })))
      .then(data => appendUniqueId(data, 'autocomplete'))
      .then(data => this.setState({ data }))
      .catch(e => console.error(e));
  };

  handleSelect = (chosen: Chosen) => {
    const { onSelect } = this.props;
    onSelect(chosen.pathLabel);
  };

  render() {
    const { data } = this.state;
    return (
      <Autocomplete
        data={data}
        onSelect={this.handleSelect}
        onChange={this.handleChange}
        filter={false}
      />
    );
  }
}

export default AutocompleteWrapper;
