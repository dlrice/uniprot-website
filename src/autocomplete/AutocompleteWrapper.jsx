// @flow
import React, { Component } from 'react';
import { Autocomplete } from 'franklin-sites';
import fetchData from '../utils/fetchData';
import appendUniqueId from './prepareData';
import { getSuggesterUrl } from '../advanced-search/utils/apiUrls';

type Props = {
  url: string,
  onSelect: Function,
};

type Suggestions = {
  dictionary: string,
  query: string,
  suggestions: Array<string>,
};

type State = {
  data: Array<Suggestions>,
};

type Chosen = {
  id: string,
  itemLabel: string,
  pathLabel: string,
};

class AutocompleteWrapper extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { data: [] };
  }

  handleChange = (textInputValue: string) => {
    const { url } = this.props;
    const suggesterUrl = getSuggesterUrl(url, textInputValue);
    this.fetchOptions(suggesterUrl);
  };

  fetchOptions = (url: string) => {
    fetchData(url)
      .then(data => data.data.suggestions.map(x => ({ pathLabel: x, itemLabel: x })))
      .then(data => appendUniqueId(data, 'autocomplete'))
      .then(data => this.setState({ data }))
      .catch(e => console.error(e));
  };

  handleSelect = (chosen: Chosen) => {
    const { onSelect } = this.props;
    onSelect(chosen.pathLabel);
  };

  render() {
    const { data } = this.state;
    return (
      <Autocomplete
        data={data}
        onSelect={this.handleSelect}
        onChange={this.handleChange}
        filter={false}
      />
    );
  }
}

export default AutocompleteWrapper;
