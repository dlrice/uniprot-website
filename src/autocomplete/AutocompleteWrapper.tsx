import React, { Component } from 'react';
import { Autocomplete } from 'franklin-sites';
import { v1 } from 'uuid';
import fetchData from '../utils/fetchData';
import appendUniqueId from './prepareData';
import { getSuggesterUrl } from '../utils/apiUrls';

type Props = {
  url: string;
  onSelect: Function;
};

type Suggestions = {
  dictionary: string;
  query: string;
  suggestions: Array<string>;
};

type State = {
  data: Array<Suggestions>;
};

type Chosen = {
  id: string;
  itemLabel: string;
  pathLabel: string;
};

class AutocompleteWrapper extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { data: [] };
    this.id = v1();
  }

  handleChange = (textInputValue: string) => {
    const { url } = this.props;
    const suggesterUrl = getSuggesterUrl(url, textInputValue);
    this.fetchOptions(suggesterUrl);
  };

  fetchOptions = (url: string) => {
    fetchData(url)
      .then(data => data.data.suggestions.map(x => ({ pathLabel: x, itemLabel: x })))
      .then(data => appendUniqueId(data, 'autocomplete'))
      .then(data => this.setState({ data }))
      .catch(e => console.error(e));
  };

  handleSelect = (chosen: Chosen) => {
    const { onSelect } = this.props;
    onSelect(chosen.pathLabel);
  };

  render() {
    const { data } = this.state;
    const { title, value } = this.props;
    return (
      <label htmlFor={this.id}>
        {title}
        <Autocomplete
          id={this.id}
          data={data}
          onSelect={this.handleSelect}
          onChange={this.handleChange}
          filter={false}
          value={value}
        />
      </label>
    );
  }
}

export default AutocompleteWrapper;
