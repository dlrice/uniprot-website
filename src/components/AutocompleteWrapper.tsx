import React, { Component } from 'react';
import { Autocomplete } from 'franklin-sites';
import { v1 } from 'uuid';
import fetchData from '../utils/fetchData';
import { getSuggesterUrl } from '../utils/apiUrls';

type Props = {
  url: string;
  onSelect: Function;
  title: string;
  value?: string;
};

type Suggestion = {
  value: string;
  id: string;
};

type Suggestions = {
  dictionary: string;
  query: string;
  suggestions: Array<Suggestion>;
};

type State = {
  data: Array<SelectValue>;
};

type SelectValue = {
  id: string;
  itemLabel: string;
  pathLabel: string;
  apiId: string;
};

class AutocompleteWrapper extends Component<Props, State> {
  id: string;
  constructor(props: Props) {
    super(props);
    this.state = { data: [] };
    this.id = v1();
  }
  static prepareData(suggestions: Suggestion[]) {
    return suggestions.map(
      (suggestion: Suggestion) =>
        ({
          pathLabel: `${suggestion.value} [${suggestion.id}]`,
          itemLabel: suggestion.value,
          apiId: suggestion.id,
          id: v1(),
        } as SelectValue)
    );
  }

  handleChange = (textInputValue: string) => {
    const { url } = this.props;
    const suggesterUrl = getSuggesterUrl(url, textInputValue);
    this.fetchOptions(suggesterUrl);
    this.handleSelect(textInputValue);
  };

  fetchOptions = (url: string) => {
    fetchData(url)
      .then(data => AutocompleteWrapper.prepareData(data.data.suggestions))
      .then(data => this.setState({ data }))
      .catch(e => console.error(e));
  };

  handleSelect = (inputValue: SelectValue | string) => {
    const { onSelect } = this.props;
    if (typeof inputValue === 'string') {
      onSelect(inputValue);
    } else {
      onSelect(inputValue.pathLabel, inputValue.apiId);
    }
  };

  render() {
    const { data } = this.state;
    const { title, value } = this.props;
    return (
      <label htmlFor={this.id}>
        {title}
        <Autocomplete
          id={this.id}
          data={data}
          onSelect={this.handleSelect}
          onChange={this.handleChange}
          filter={false}
          value={value}
        />
      </label>
    );
  }
}

export default AutocompleteWrapper;
