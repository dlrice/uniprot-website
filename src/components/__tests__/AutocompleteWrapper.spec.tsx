import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import AutocompleteWrapper from '../AutocompleteWrapper';
import { resetUuidV1 } from '../../../__mocks__/uuid';
import {
  suggesterMock,
  preparedSuggestions,
} from '../__fixtures__/AutocompleteWrapper';

describe('Autocomplete Wrapper static methods', () => {
  test('should prepare API data for Autocomplete', () => {
    resetUuidV1();
    expect(
      AutocompleteWrapper.prepareData(suggesterMock.response.suggestions)
    ).toEqual(preparedSuggestions);
  });
});

const props = {
  title: 'Taxonomy [OC]',
  value: 'Homo sapiens (Human) [9606]',
  inputValue: suggesterMock.query,
  url: suggesterMock.baseUrl,
  onSelect: jest.fn(),
};
const mock = new MockAdapter(axios);
mock.onGet().reply(200, suggesterMock.response);

let rendered;
describe('Autocomplete Wrapper', () => {
  beforeEach(() => {
    resetUuidV1();
    rendered = render(<AutocompleteWrapper {...props} />);
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the correct number of AutocompleteItems when input is human', async () => {
    const { queryByTestId, getAllByTestId } = rendered;
    const searchInput = queryByTestId('search-input');
    fireEvent.change(searchInput, { target: { value: suggesterMock.query } });
    const autocompleteItems = await waitForElement(() =>
      getAllByTestId('autocomplete-item')
    );
    expect(autocompleteItems.length).toEqual(
      suggesterMock.response.suggestions.length
    );
  });

  test('should not render AutocompleteItems when input is less than minCharsToShowDropdown (=3)', async () => {
    const { queryByTestId, queryAllByTestId } = rendered;
    const searchInput = queryByTestId('search-input');
    const value = 'hu';
    fireEvent.change(searchInput, { target: { value } });
    const autocompleteItems = await waitForElement(() =>
      queryAllByTestId('autocomplete-item')
    );
    expect(autocompleteItems.length).toEqual(0);
  });
});
