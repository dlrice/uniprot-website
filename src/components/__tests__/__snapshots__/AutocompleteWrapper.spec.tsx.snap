// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete Wrapper should render 1`] = `
"<label htmlFor={0}>
  Taxonomy [OC]
  <e id={0} data={{...}} onSelect={[Function]} onChange={[Function]} filter={false} value=\\"Homo sapiens (Human) [9606]\\" showDropwdownUpdated={[Function: showDropwdownUpdated]} clearOnSelect={false} placeholder=\\"\\" />
</label>"
`;
